plugin/evm/acp176/acp176_test.go

```diff
@@ -0,0 +1,694 @@
+					Capacity: 1_000_000,
+					Excess:   2_000_000,
+				},
+			},
+			gasUsed:      1_000_001,
+			extraGasUsed: nil,
+			expectedErr:  gas.ErrInsufficientCapacity,
+			expected:     State{},
+		},
+		{
+			name: "massive_extra_gas_used",
+			initial: State{
+				Gas: gas.State{
+					Capacity: 1_000_000,
+					Excess:   2_000_000,
+				},
+			},
+			gasUsed:      0,
+			extraGasUsed: new(big.Int).Lsh(common.Big1, 64),
+			expectedErr:  gas.ErrInsufficientCapacity,
+			expected: State{
+				Gas: gas.State{
+					Capacity: 1_000_000,
+					Excess:   2_000_000,
+				},
+			},
+		},
+		{
+			name: "extra_gas_used_capacity_exceeded",
+			initial: State{
+				Gas: gas.State{
+					Capacity: 1_000_000,
+					Excess:   2_000_000,
+				},
+			},
+			gasUsed:      0,
+			extraGasUsed: big.NewInt(1_000_001),
+			expectedErr:  gas.ErrInsufficientCapacity,
+			expected:     State{},
+		},
+	}
+	updateTargetExcessTests = []struct {
+		name                string
+		initial             State
+		desiredTargetExcess gas.Gas
+		expected            State
+	}{
+		{
+			name: "no_change",
+			initial: State{
+				Gas: gas.State{
+					Excess: 2_000_000,
+				},
+				TargetExcess: 0,
+			},
+			desiredTargetExcess: 0,
+			expected: State{
+				Gas: gas.State{
+					Excess: 2_000_000,
+				},
+				TargetExcess: 0,
+			},
+		},
+		{
+			name: "max_increase",
+			initial: State{
+				Gas: gas.State{
+					Excess: 2_000_000,
+				},
+				TargetExcess: 0,
+			},
+			desiredTargetExcess: MaxTargetExcessDiff + 1,
+			expected: State{
+				Gas: gas.State{
+					Excess: 2_001_954, // 2M * NewTarget / OldTarget
+				},
+				TargetExcess: MaxTargetExcessDiff, // capped
+			},
+		},
+		{
+			name: "inverse_max_increase",
+			initial: State{
+				Gas: gas.State{
+					Excess: 2_001_954,
+				},
+				TargetExcess: MaxTargetExcessDiff,
+			},
+			desiredTargetExcess: 0,
+			expected: State{
+				Gas: gas.State{
+					Excess: 2_000_000, // inverse of max_increase
+				},
+				TargetExcess: 0,
+			},
+		},
+		{
+			name: "max_decrease",
+			initial: State{
+				Gas: gas.State{
+					Excess: 2_000_000_000,
+				},
+				TargetExcess: 2 * MaxTargetExcessDiff,
+			},
+			desiredTargetExcess: 0,
+			expected: State{
+				Gas: gas.State{
+					Excess: 1_998_047_816, // 2M * NewTarget / OldTarget
+				},
+				TargetExcess: MaxTargetExcessDiff,
+			},
+		},
+		{
+			name: "inverse_max_decrease",
+			initial: State{
+				Gas: gas.State{
+					Excess: 1_998_047_816,
+				},
+				TargetExcess: MaxTargetExcessDiff,
+			},
+			desiredTargetExcess: 2 * MaxTargetExcessDiff,
+			expected: State{
+				Gas: gas.State{
+					Excess: 1_999_999_999, // inverse of max_decrease -1 due to rounding error
+				},
+				TargetExcess: 2 * MaxTargetExcessDiff,
+			},
+		},
+	}
+)
+
+func TestTarget(t *testing.T) {
+	for _, test := range readerTests {
+		t.Run(test.name, func(t *testing.T) {
+			require.Equal(t, test.target, test.state.Target())
+		})
+	}
+}
+
+func BenchmarkTarget(b *testing.B) {
```

Comment:
What's the objective of these benchmarks? 👀 🤔 
Are we comparing i.e. two implementations?

Reply:
The benchmarks are here to measure the performance of the package. Many of these functions are called during block processing and are therefore on a critical consensus path. They can't be slow.

Reply:
> I don't think a future developer would know what to think of these benchmarks.

I think a future developer should be capable of looking at the performance difference a PR makes to determine if the change is acceptable or not.